account_invoice_pricelist/models/account_invoice.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    pricelist_id = fields.Many2one(
+        comodel_name='product.pricelist', string='Pricelist',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+
+    @api.onchange('partner_id', 'company_id')
+    def _onchange_partner_id_account_invoice_pricelist(self):
+        result = super(AccountInvoice, self)._onchange_partner_id()
+        if self.partner_id:
+            if self.type in ('out_invoice', 'out_refund')\
+                    and self.partner_id.property_product_pricelist:
+                # Customer Invoices
+                self.pricelist_id = self.partner_id.property_product_pricelist
+            elif self.type in ('in_invoice', 'in_refund')\
```

Comment:
There are no more purchase pricelists, so this should be removed.

Reply:
ok

Reply:
@pedrobaeza ok to your remarks. But about squashing the commits. Do you think about specific commits? I think I mostly give useful details about what comes from v9, what comes from v8 (because both evolved independently) and also it is not so easy to squash commits because of the merge of the account_invoice_line into the account_invoice file. So can we forget that or may be you give me specific commits I can squash. I'll amend my last commit to take your last remarks into account.